soc/arm/nordic_nrf/common/soc_secure.c

```diff
@@ -44,4 +49,7 @@ int soc_secure_mem_read(void *dst, void *src, size_t len)
 	default:
 		return -EPERM;
 	}
+#else
+	return -ENOTSUP;
+#endif
```

Comment:
There is no code-reuse in this file between CONFIG_TFM_PARTITION_PLATFORM and not CONFIG_TFM_PARTITION_PLATFORM.

I suggest we add a soc_secure_dummy.c file to simplify the code of the common case.

Reply:
Change reverted.